app/service/event_svc.py

```diff
@@ -0,0 +1,35 @@
+import asyncio
+import json
+import websockets
+
+from app.utility.base_service import BaseService
+
+
+class EventService(BaseService):
+
+    def __init__(self):
+        self.log = self.add_service('event_svc', self)
+        self.contact_svc = self.get_service('contact_svc')
+        self.ws_uri = 'ws://{}'.format(self.get_config('app.contact.websocket'))
+
+    async def observe_event(self, event, callback):
+        ws_contact = [c for c in self.contact_svc.contacts if c.name == 'websocket']
+        handle = _Handle(event, callback)
+        ws_contact[0].handler.handles.append(handle)
+
+    async def fire_event(self, event, **callback_kwargs):
+        uri = '{}/{}'.format(self.ws_uri, event)
+        msg = json.dumps(callback_kwargs)
+        loop = asyncio.get_event_loop()
+        async with websockets.connect(uri) as websocket:
+            loop.create_task(websocket.send(msg))
```

Comment:
I wouldn't make the loop variable at line 23. at line 25, I would just do: asyncio.get_event_loop().create_task(...)

in case there is no connection at line 24, we can avoid creating the variable that isn't really necessary. bonus, 1 line saved ;)